### ISSUE TITLE
Compilation error with angular 7.1.3 and buildOptimizer

### ISSUE THREAD

**Issue by martinwoitzik** [stomp-js/ng2-stompjs#126]
Hey guys,

there already was a bug related to ng7.0.x which is exactly the problem we currently encounter with ng7.1.3: https://github.com/stomp-js/ng2-stompjs/issues/103

When compiling with --prod the page stays white and this error occurs:
`main.aa1837ccac249ca777f9.js:1 Uncaught TypeError: Object prototype may only be an Object or null: undefined
    at setPrototypeOf (<anonymous>)`

Do you have any solution for that? Setting buildOptimizer to false works but that's not really a solution.

Cheers,
Martin

**Comment by martinwoitzik** [comment]
We tried it with angular 7.2.0 and your beta.1 version today and still have the same issue during runtime. :(
Any idea what we could do to fix it?

**Comment by martinwoitzik** [comment]
We also did some investigation. Starting our app with "ng serve --prod" (which seems to somehow use the optimizer but without uglification) shows that there is a null-pointer regarding RxStompState, which seems to be undefined.

This is the runtime-error:
`main.981c1ecf6c7cfa5e02af.js:30268 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'CLOSED' of undefined
TypeError: Cannot read property 'CLOSED' of undefined
    at RxStompService.RxStomp (main.981c1ecf6c7cfa5e02af.js:70259)
    at new RxStompService (main.981c1ecf6c7cfa5e02af.js:69406)`

And if you look inside main.js:
`
            var RxStomp = /*@__PURE__*/ (function () {
                /**
                 * Constructor
                 */
                function RxStomp() {
                    var _this = this;
                    /**
                     * Internal array to hold locally queued messages when STOMP broker is not connected.
                     */
                    this._queuedMessages = [];
                    this._stompClient = new stompjs_1.Client();
                    var noOp = function () { };
                    // Before connect is no op by default
                    this._beforeConnect = noOp;
                    // debug is no-op by default
                    this._debug = noOp;
                    // Initial state is CLOSED
                    this._connectionStatePre$ = new rxjs_1.BehaviorSubject(rx_stomp_state_1.RxStompState.CLOSED);
                    this._connectedPre$ = this._connectionStatePre$.pipe(operators_1.filter(function (currentState) {
                        return currentState === rx_stomp_state_1.RxStompState.OPEN;
                    }));`

... it throws a null-pointer expection in the line where "RxStompState.CLOSED" is used

Looking at your packaged code, we also discovered that you don't export the "RxStompState" globally (looking inside "index.js" there is only "StompState" available). 
It would be great to use the RxStompStates also in our source code instead of using magic numbers. So maybe if you fix the export issue, the problem may also disappear?

**Comment by kum-deepak** [comment]
`RxStompState` is exposed in `@stomp/rx-stomp`. The optimizer might have optimized it away (which itself may not be an error).

I have created a fresh new Angular App using newest `@angular/cli`, added `@stomp/ng2-stompjs`, copied the `app` folder from previous sample, and executed `ng serve --prod --optimization`. The application seems to be working fine.

```bash
$ npm i -g @angular/cli
$ ng new myapp
$ cd myapp
$ npm i
$ npm i @stomp/ng2-stompjs
$ # Copy App folder from previous demo
$ ng serve --prod --optimization
```

The above is available at branch `angular-7.2` of https://github.com/stomp-js/ng2-stompjs-angular7. This sample uses `RxStompState` to show the status as text (see the status component).

My suggestions:

- Checkout branch `angular-7.2` of https://github.com/stomp-js/ng2-stompjs-angular7 to see if it works for you. 
- Please check your version of `@stomp/rx-stomp`, it should be at `0.2.0-beta.1`. Work around for #103 involved changes in packaging of `@stomp/rx-stomp`. Uninstalling both `@stomp/rx-stomp` and `@stomp/ng2-stompjs` followed by installing only `@stomp/ng2-stompjs` should correct the versions.
- If that does not work, try creating a new project, move only the app folder from your current application. Install all your dependencies by using `npm i/yarn` (i.e., do not copy your package.json or the lock). This has helped me in past in similar situations.

**Comment by kum-deepak** [comment]
Closing because lack of activity.
